test/xref2/module_preamble.t/run.t

```diff
@@ -13,6 +13,8 @@ and that "hidden" modules (eg. `A__b`, rendered to `html/A__b`) are not rendered
 
   $ odoc compile --pkg test -o a__b.odoc -I . a__b.cmti
   $ odoc compile --pkg test -o a.odoc -I . a.cmti
+  File "a.mli", line 4, characters 4-17:
+  Unexpected tag '@canonical' at this location.
```

Comment:
This isn't expected however. This is caused by `` `Root `` paths to not be valid canonical paths, I'll add a different warning for this case.

Reply:
I've just added a specific warning for this case.